service-base/src/main/java/org/eclipse/hono/service/resourcelimits/ResourceLimitChecks.java

```diff
@@ -31,9 +32,26 @@
      *         <p>
      *         The future will be failed with a {@link ServiceInvocationException}
      *         if the check could not be performed.
+     * @deprecated Use {@link #isConnectionLimitReached(TenantObject, Span)} instead.
      */
+    @Deprecated
     Future<Boolean> isConnectionLimitReached(TenantObject tenantObject);
 
+    /**
+     * Checks if the maximum number of connections configured for a tenant
+     * have been reached.
+     *
+     * @param tenantObject The tenant configuration to check the limit against.
+     * @param span The active OpenTracing span for this operation. It is not to be closed in this method!
+     *             An implementation should log (error) events on this span, and it may set tags and use 
+     *             this span as the parent for any spans created in this method.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
+     */
+    Future<Boolean> isConnectionLimitReached(TenantObject tenantObject, Span span);
```

Comment:
FMPOV we should pass in a `SpanContext` instead of a `Span`. It should be up to the implementation whether it wants to trace the execution of the check using a new (child) span or not.

Reply:
I created a child span in the abstract base class and passed it to the resource limit checks. I intended to make it as a child span as it looks more elegant and easier to track the flow in this case. In Hono currently I see two approaches. One is passing the `SpanContext`and the other is passing the `Span` as in the `TenantManagementService`. I chose the `Span` in the above implementation. IMHO passing the `Span` shouldn't prevent the implementation to create any new child spans if needed, as `Span.context()` returns the `SpanContext`.

Reply:
That is true but it also allows the method to finish the span. FMPOV it is more about the question who is in control of and responsible for the life-cycle of a span. In this case, my feeling is that the responsibility should lie with the `ResourceLimitChecks` implementation. If the `AbstractProtocolAdapterBase` class creates a Span, then it should either be in control of the span's life-cycle itself (meaning that it only passes a SpanContext to the invoked methods of `ResourceLimitChecks`) or the `ResourceLimitChecks` interface JavaDoc needs to be very specific about the expectation, how implementations should handle the passed in Span. Personally, I believe that passing in a SpanContext only is much safer and provides for better encapsulation of the Span life-cycle.

Reply:
> In this case, my feeling is that the responsibility should lie with the `ResourceLimitChecks` implementation.

I think so too that the responsibility should lie with the `ResourceLimitChecks` and I will pass the `SpanContext` instead of `Span`.